Gems/PhysX/Code/Editor/ColliderComponentMode.cpp

```diff
@@ -159,10 +160,18 @@ namespace PhysX
 
     void ColliderComponentMode::SetCurrentMode(SubMode newMode)
     {
-        AZ_Assert(m_subModes.count(newMode) > 0, "Submode not found:%d", newMode);
+        auto subMode = m_subModes.find(newMode);
+        AZ_Assert(subMode != m_subModes.end(), "Submode not found:%d", newMode);
```

Comment:
Since `subMode` is only used in the assert this will fail build in release. Since the `find` is not necessary in release, I would put it inside the assert itself instead of using `[[maybe_unused]]`.

````
AZ_Assert(m_subModes.find(newMode) != m_subModes.end(), "Submode not found:%d", newMode);
````

Reply:
Good catch! Fixed